### ISSUE TITLE
Relational: Incorrect column binding with the same column name

### ISSUE THREAD

**Issue by ravengerUA** [aspnet/EntityFramework#2554]
We have two entities with the same column name (eg. Name)

Entities
````sql
[Table("award")]
public class Award
{
    [Column("award_id")]
    public int Id { get; set; }

    [Column("name")]
    public string Name { get; set; }

    [Column("country_id")]
    public int CountryId { get; set; }
    public Country Country { get; set; }
}

[Table("countries")]
public class Country
{
    [Column("country_id")]
    public int Id { get; set; }

    [Column("name")]
    public string Name { get; set; }
}
````
and we have this code
````
from a in db.Awards
where a.Id == 1
join c in db.Countries on a.CountryId equals c.Id
select new
{
    AwardId = a.Id,
    Name = a.Name,
    CountryId = c.Id,
    CountryName = c.Name
};
````
We get an error
````
The multi-part identifier "c.country_id" could not be bound.
The multi-part identifier "c.name" could not be bound.
````
But if we write "where a.Id == 1" after inner join, then the code starts work

EF 6.x doesn't have this bug.

**Comment by ErikEJ** [comment]
beta4, beta5 or latest nightly?

**Comment by ravengerUA** [comment]
The latest nightly (7.0.0-beta6-13679)

**Comment by rowanmiller** [comment]
@ravengerUA could you share the SQL that is being sent to the database?

**Comment by ravengerUA** [comment]
This SQL
````sql
SELECT TOP(1) [a].[award_id], [a].[name], [c].[country_id], [c].[name]
FROM [awards] AS [a]
WHERE [a].[award_id] = @__awardId_0
````

**Comment by maumar** [comment]
Fixed in 45eed86223434eb180b81ec1074e3abb36e78778
